Docs: Add Precision Support Page

adeljo-amd opened the pull request:
Adds a precision support page to rocSPARSE's documentation

adeljo-amd committed:
Docs: Add Precision Support Page

amd-jnovotny commented on docs/reference/precision.rst:
```diff
@@ -0,0 +1,108 @@
+    :header-rows: 1
+
+    *
+      - Type Prefix
+      - C++ Type
```
```suggestion
      - Type prefix
      - C++ type
```

amd-jnovotny commented on docs/reference/precision.rst:
```diff
@@ -0,0 +1,108 @@
+-----------------------------
+
+Most sparse matrix operations in hipSPARSE support both real and complex precisions, however certain algorithms
+or optimizations may be specific to real or complex data:
```
```suggestion
Most sparse matrix operations in hipSPARSE support both real and complex precisions, however, certain algorithms
or optimizations might be specific to real or complex data:
```

amd-jnovotny commented on docs/reference/precision.rst:
```diff
@@ -0,0 +1,108 @@
+Most sparse matrix operations in hipSPARSE support both real and complex precisions, however certain algorithms
+or optimizations may be specific to real or complex data:
+
+* Some functions may have different performance characteristics for complex data compared to real data.
+* Certain conversion routines may operate differently depending on whether the data is real or complex.
```
```suggestion
* Some functions might have different performance characteristics for complex data compared to real data.
* Certain conversion routines might operate differently depending on whether the data is real or complex.
```

amd-jnovotny commented on docs/reference/precision.rst:
```diff
@@ -0,0 +1,108 @@
+--------------------
+
+The four core precision types (s, d, c, z) are supported across most functions in hipSPARSE. Some specialized
+functions may only support a subset of these types. Always check the specific function documentation to confirm
```
```suggestion
functions might only support a subset of these types. See the specific function documentation to confirm
```

amd-jnovotny reviewed (approved):
Looks good. Just a handful of suggestions

adeljo-amd commented:
@amd-jnovotny Thanks! I've made the changes - the PR on rocSPARSE is pretty much the same as this PR so I've made the same fixes there

amd-jnovotny commented:
Thanks @adeljo-amd. Looks good now. I also approved the rocSPARSE PR.

adeljo-amd commented:
@amd-jnovotny Thanks! Should we get another set of eyes on this, or would you rather I merge this in given that you're currently refreshing the docs structure here?

amd-jnovotny commented:
@adeljo-amd: Probably best to get the dev team to take a quick look, then you can merge after that.

adeljo-amd commented:
@jsandham Would you have some free time to review this PR and the [PR](https://github.com/ROCm/rocSPARSE/pull/447) over at rocSPARSE? It would be great if you could provide some feedback on these from a developer perspective

jsandham commented:
> @jsandham Would you have some free time to review this PR and the [PR](https://github.com/ROCm/rocSPARSE/pull/447) over at rocSPARSE? It would be great if you could provide some feedback on these from a developer perspective

I built it locally today and it looks good. One minor point. Both hipSPARSE and rocSPARSE have what we refer to as "generic" routines, i.e. hipsparseSpMV etc. These routines do not follow the convention of using s, d, c or z to indicate their precision. Instead these generic routines are more flexible and allow for mixed precisions. I think we may want to mention this in the precision page. I can help with the word here if needed. In the meantime, can you resolve the merge conflicts that arose after the docs refactor was merged?

adeljo-amd commented:
> > @jsandham Would you have some free time to review this PR and the [PR](https://github.com/ROCm/rocSPARSE/pull/447) over at rocSPARSE? It would be great if you could provide some feedback on these from a developer perspective
> 
> I built it locally today and it looks good. One minor point. Both hipSPARSE and rocSPARSE have what we refer to as "generic" routines, i.e. hipsparseSpMV etc. These routines do not follow the convention of using s, d, c or z to indicate their precision. Instead these generic routines are more flexible and allow for mixed precisions. I think we may want to mention this in the precision page. I can help with the word here if needed. In the meantime, can you resolve the merge conflicts that arose after the docs refactor was merged?

I've rebased and added a short section about generic routines - let me know if this works. If it does, I'll replicate this in the other PR (since both PRs share the same structure)

jsandham commented on docs/reference/precision.rst:
```diff
@@ -0,0 +1,145 @@
+
+This approach enables:
+
+* Using different precision types for matrices and vectors.
```
Could change this to:

"_Using different precision types for matrices and vectors. The advantage of using different data types is to save on memory bandwidth and storage when a user application allows while performing the actual computation in a higher precision_"

This helps explain why a user might want to do this. I am fine with not including this as well. Ill defer to your expertise on this. Otherwise looks good. Approved here and in the other PR's.

jsandham replied:
Since this is also related to the point below, you could also just add this addtional comment at the end and a non-bullet point sentence.

jsandham replied:
Something like:

* Using different precision types for matrices and vectors.
* Specifying computation precision independently of storage precision.
* Supporting mixed precision workflows with a unified API.

The advantage of using different data types is to save on memory bandwidth and storage when a user application allows while performing the actual computation in a higher precision

adeljo-amd replied:
I like the last one, so I integrated that into this PR and the other PR
